src/cmap/auth/mongodb_oidc/azure_service_workflow.ts

```diff
@@ -0,0 +1,96 @@
+import { MongoAzureError } from '../../../error';
+import { request } from '../../../utils';
+import { AzureTokenCache } from './azure_token_cache';
+import { ServiceWorkflow } from './service_workflow';
+
+/** Error for when the token audience is missing in the environment. */
+const TOKEN_AUDIENCE_MISSING_ERROR = 'TOKEN_AUDIENCE must be set in the environment.';
+
+/** Base URL for getting Azure tokens. */
+const AZURE_BASE_URL =
```

Comment:
Once we bring CSFLE into the driver we can move these constants into a common Azure shared module.

Reply:
Sonuds good, wanna add a bullet to the FLE ticket calling that out, just something simple as "consolidate constants" would make us remember to go looking for dupes. 

We also have an azure token cache in FLE ( cc: @baileympearson ) allegedly there could be a way to combine these as well.